tests: net: no_compat_namespacing_openthread: frdm_rw612: build failure

### Describe the bug

build failure for net.build.no_compat_namespacing_openthread

platform:

- frdm_rw612

f97e33eb387491dd94d07b8ec6b6a744c589c09b is the first bad commit
commit f97e33eb387491dd94d07b8ec6b6a744c589c09b
Author: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
Date:   Tue Nov 18 16:27:18 2025 +0200

    tests: net: all: Verify OpenThread compilation
    
    Try to catch various OpenThread related compilation issues.
    
    Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

 tests/net/all/testcase.yaml | 90 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 90 insertions(+)

### Regression

- [x] This is a regression.

### Steps to reproduce

west twister -p frdm_rw612 -T test/net/all 

### Relevant log output

```shell
[129/210] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_mcg.c.obj
[130/210] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_sim.c.obj
[131/210] Linking C static library zephyr/drivers/clock_control/libdrivers__clock_control.a
[132/210] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj
[133/210] Linking C static library zephyr/drivers/console/libdrivers__console.a
[134/210] Building C object zephyr/drivers/entropy/CMakeFiles/drivers__entropy.dir/entropy_mcux_trng.c.obj
[135/210] Linking C static library zephyr/drivers/entropy/libdrivers__entropy.a
[136/210] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_mcux.c.obj
[137/210] Linking C static library zephyr/drivers/gpio/libdrivers__gpio.a
[138/210] Building C object zephyr/drivers/ieee802154/CMakeFiles/drivers__ieee802154.dir/ieee802154_kw41z.c.obj
FAILED: zephyr/drivers/ieee802154/CMakeFiles/drivers__ieee802154.dir/ieee802154_kw41z.c.obj -format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/net/lib/mqtt_sn_packet=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c17 -MD -MT zephyr/drivers/ieee802154/CMakeFiles/drivers__ieee802154.dir/ieee802154_kw41z.c.obj -MF zephyr/drivers/ieee802154/CMakeFiles/drivers__ieee802154.dir/ieee802154_kw41z.c.obj.d -o zephyr/drivers/ieee802154/CMakeFiles/drivers__ieee802154.dir/ieee802154_kw41z.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ieee802154/ieee802154_kw41z.c/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ieee802154/ieee802154_kw41z.c: 
In function 'handle_ack':/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ieee802154/ieee802154_kw41z.c:584:48: 
error: 'AF_UNSPEC' undeclared (first use in this function); did you mean 'NET_AF_UNSPEC'? 584 | AF_UNSPEC, 0, K_NO_WAIT);
 | 		^~~~~~~~~ | NET_AF_UNSPEC/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ieee802154/ieee802154_kw41z.c:584:48:
 note: each undeclared identifier is reported only once for each function it appears inninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- toolchain: Zephyr
- version:  v4.3.0-912-g9463d9a51d9c

### Additional Context

_No response_

--- a/drivers/ieee802154/ieee802154_esp32.c
+++ b/drivers/ieee802154/ieee802154_esp32.c
@@ -232,7 +232,7 @@ static int handle_ack(struct ieee802154_esp32_data *data)
 		ack_len = data->ack_frame[0] - IEEE802154_FCS_LENGTH;
 	}
 
-	ack_pkt = net_pkt_rx_alloc_with_buffer(data->iface, ack_len, AF_UNSPEC, 0, K_NO_WAIT);
+	ack_pkt = net_pkt_rx_alloc_with_buffer(data->iface, ack_len, NET_AF_UNSPEC, 0, K_NO_WAIT);
 	if (!ack_pkt) {
 		LOG_ERR("No free packet available.");
 		err = -ENOMEM;

--- a/drivers/ieee802154/ieee802154_kw41z.c
+++ b/drivers/ieee802154/ieee802154_kw41z.c
@@ -581,7 +581,7 @@ static void handle_ack(struct kw41z_context *kw41z, uint8_t seq_number)
 	uint8_t ack_psdu[ACK_FRAME_LEN];
 
 	ack_pkt = net_pkt_rx_alloc_with_buffer(kw41z->iface, ACK_FRAME_LEN,
-					       AF_UNSPEC, 0, K_NO_WAIT);
+					       NET_AF_UNSPEC, 0, K_NO_WAIT);
 	if (!ack_pkt) {
 		LOG_ERR("No free packet available.");
 		return;

